Python/omevv/v1/scripts/GetRepoDetails.py

```diff
@@ -0,0 +1,196 @@
+            response = requests.get(self.repo_url, headers=self.headers, verify=False)
+            data = response.json();
+            status_code = response.status_code;
+            print("status code in repoprof ", status_code)
+            if status_code == 200:
+                for prof in data:
+                    s = json.dumps(prof)
+                    profobj = json.loads(s, object_hook=repoProfileDecoder)
+                    prof_list.append(profobj)
+            else:
+                raise Exception("Error Occured while fetching the repo profile details ",response)
+
+        except Exception as e:
+            print("Exception occured while creating repo ", e, " retrying ..");
+            if self.retry > 0:
+                self.retry = self.retry - 1
+                time.sleep(5)
+                self.get_repoprof_details(console_ip,console_hostname,console_username,console_domain,console_pwd)
+
+            else:
+                print("Failed after 3 retries,exiting")
+                sys.exit()
+        return prof_list
+
+
+
+
+
+
+    def write_to_csv(self,listOfObjects,fileName):
+        with open(fileName, "w") as stream:
+            writer = csv.writer(stream)
+            writer.writerows(listOfObjects)
+
+        print("Successfully written to csv file")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Python script to fetch repository profile details from OMIVV"
+    );
+    parser.add_argument('-ip', help='OMIVV IP address', required=True);
+    parser.add_argument('-user', help='OMIVV username', required=True);
+    parser.add_argument('-pswd', help='OMIVV password', required=True);
+    parser.add_argument('-domain', help='OMIVV domain', required=False, default="");
+    parser.add_argument('-cip', help='console ip', required=False);
+    parser.add_argument('-cname', help='console name', required=False);
+    parser.add_argument('-cuser', help='console username', required=False);
+    parser.add_argument('-cpwd', help='console password', required=False);
+    parser.add_argument('-cdomain', help='console domain', required=False);
+    args = vars(parser.parse_args());
+    if args['cname'] == "" and args['cip'] == "":
+        raise Exception('Console ip or Console Name is required.Please pass one of them');
+
+    omevv_ip = args['ip'];
+    omevv_user = args['user'];
+    omevv_pswd = args['pswd'];
+    omevv_domain = args['domain'];
+    console_username = args['cuser']
+    console_domain = args['cdomain']
+    console_pwd = args['cpwd']
+    console_ip = args['cip']
+    console_hostname = args['cname']
+
+    repoObj = RepoDetails(omevv_ip,omevv_user,omevv_pswd,omevv_domain)
+    #repoObj.get_console_details()
+   # repoList = repoObj.get_repoprof_details(console_ip,console_hostname,"",console_domain,console_pwd)
+   # repoObj.write_to_csv(repoList,"repoprof.csv")
+    consolelist = repoObj.get_console_details()
```

Comment:
Is this code complete? What are we accomplishing by querying the list of consoles?